### DISCUSSION TITLE
remove flyway and keep final ddl schema sql file.

### DISCUSSION THREAD

**Discussion by wolfboys** [apache/streampark#481] [General]
hi:

 Since streamx uses flyway to keep the version of ddl sql schema smoothly upgraded, but:

1.  sometimes we need to modify the historical  sql ddl file, which will bring some problems
2. more often, ddl will be executed by DBA. normal user don't have permission to execute ddl, which will also cause problems

so. we need discuss whether flyway need to be removed. and keep a final ddl schema sql file.

-----
大家好:

由于streamx使用了flyway来保持版本平稳升级，这样非常方便,自动升级sql ddl, 但是：

1. 有时我们需要修改历史sql ddl文件，这会带来一些问题

2. 通常情况下ddl sql只有DBA才用权限去执行,普通用户没有执行ddl的权限，这会导致flyway自动升级失败,也是问题

所以我们需要讨论在这个阶段是否需要去掉flyway自动升级的方案, 并保留最终的ddl sql文件由用户去手动执行.

**Comment by zhongqishang** [comment]
+1 remove flyway

1. 需要学习成本
2. 初次接入用户不需要关心之前有过多少次升级
3. 没有执行ddl的权限

**Reply by monrg** [reply]
+1 remove flyway
1、保留最终的ddl sql文件更清晰

**Comment by Al-assad** [comment]
I would prefer StreamX to remove flyway at this stage to reduce the development burden. From the current community feedback, most users don't have strong demand for smooth upgrade. The way of flyway takes over the db schema ddl  bringing extra burden to the user's production deployment.

我更赞成 StreamX 在当前这个阶段移除掉 flyway，以降低开发负担。从目前的社区反馈上看，大部分用户没有平滑升级的诉求，而 flyway 这种应用接管 db schema ddl 的方式，反而给用户的生产部署带来了额外的负担。

**Comment by wolfboys** [comment]
+1
